win32ss/gdi/gdi32/objects/bitmap.c

```diff
@@ -731,29 +729,6 @@ SetDIBitsToDevice(
 
     cjBmpScanSize = DIB_BitmapMaxBitsSize((LPBITMAPINFO) lpbmi, ScanLines);
 
-    pvSafeBits = RtlAllocateHeap(GetProcessHeap(), 0, cjBmpScanSize);
```

Comment:
This sort of reallocation + copy might be needed only if the Bits pointer is not DWORD-aligned....

Reply:
Yep, except our NtGdiSetDIBits doesn't require alignment, either.

Reply:
*our*, perhaps... but then what would happen if you put our gdi32.dll with MS' win32k.sys? :trollface: